python/servo/command_base.py

```diff
@@ -388,6 +392,77 @@ def get_binary_path(self, release, dev, android=False):
                                   " --release" if release else ""))
         sys.exit()
 
+    def get_nightly_binary_path(self, nightly_date):
+        if nightly_date is None:
+            return
+        if not nightly_date:
+            print(
+                "No nightly date has been provided although the --nightly or -n flag has been passed.")
+            sys.exit(1)
+        # Will alow us to fetch the relevant builds from the nightly repository
+        os_prefix = "linux"
+        if is_windows():
+            print("The nightly flag is not supported on windows yet.")
+            sys.exit(1)
+        if is_macosx():
+            print("The nightly flag is not supported on mac yet.")
+            sys.exit(1)
+
+        # Fetch the filename to download from the build list
+        repository_index = NIGHTLY_REPOSITORY_URL + "?list-type=2&prefix=nightly"
+        req = urllib2.Request(
+            "{}/{}/{}".format(repository_index, os_prefix, nightly_date))
+        try:
+            response = urllib2.urlopen(req).read()
+            tree = XML(response)
+            namespaces = {'ns': tree.tag[1:tree.tag.index('}')]}
+            file_to_download = tree.find('ns:Contents', namespaces).find(
+                'ns:Key', namespaces).text
+        except urllib2.URLError as e:
+            print("Could not fetch the available nightly versions from the repository : {}".format(
+                e.reason))
+            sys.exit(1)
+        except AttributeError as e:
+            print("Could not fetch a nightly version for date {} and platform {} ".format(
+                nightly_date, os_prefix))
+            sys.exit(1)
+
+        nightly_target_directory = path.join(self.context.topdir, "target")
+        # Once extracted, the nightly folder name is the tar name without the extension
+        # (eg /foo/bar/baz.tar.gz extracts to /foo/bar/baz)
+        destination_file = path.join(
+            nightly_target_directory, file_to_download)
+        destination_folder = destination_file[:destination_file.index('.')]
```

Comment:
We can use `os.path.splitext(destination_file)[0]` instead.

Reply:
I didn't know about this method, pretty cool !